.github/workflows/build_and_unitest.yml

```diff
@@ -9,9 +9,103 @@ on:
+        id: changed-files-interfacing
+        uses: tj-actions/changed-files@v42
+        with:
+          files: src/interfacing/**
+      
+      - name: Get changed files inside perception folder
+        id: changed-files-perception
+        uses: tj-actions/changed-files@v42
+        with:
+          files: src/perception/**
+      
+      - name: Get changed files inside samples folder
+        id: changed-files-samples
+        uses: tj-actions/changed-files@v42
+        with:
+          files: src/samples/**
+      
+      - name: Get changed files inside simulation folder
+        id: changed-files-simulation
+        uses: tj-actions/changed-files@v42
+        with:
+          files: src/simulation/**
+      
+      - name: Get changed files inside world_modeling folder
+        id: changed-files-world-modeling
+        uses: tj-actions/changed-files@v42
+        with:
+          files: src/world_modeling/**
+
+      - name: Update CHANGED_MODULES for action
+        if: steps.changed-files-action.outputs.any_changed == 'true'
+        run: |
+          echo "Detected action changes"
+          echo "CHANGED_MODULES=$CHANGED_MODULES action" >> $GITHUB_ENV
+
+      - name: Update CHANGED_MODULES for interfacing
+        if: steps.changed-files-interfacing.outputs.any_changed == 'true'
+        run: |
+          echo "Detected interfacing changes"
+          echo "CHANGED_MODULES=$CHANGED_MODULES interfacing" >> $GITHUB_ENV
+
+      - name: Update CHANGED_MODULES for perception
+        if: steps.changed-files-perception.outputs.any_changed == 'true'
+        run: |
+          echo "Detected perception changes"
+          echo "CHANGED_MODULES=$CHANGED_MODULES perception" >> $GITHUB_ENV
+        
+      - name: Update CHANGED_MODULES for samples
+        if: steps.changed-files-samples.outputs.any_changed == 'true'
+        run: |
+          echo "Detected samples changes"
+          echo "CHANGED_MODULES=$CHANGED_MODULES samples" >> $GITHUB_ENV
+
+      - name: Update CHANGED_MODULES for simulation
+        if: steps.changed-files-simulation.outputs.any_changed == 'true'
+        run: |
+          echo "Detected simulation changes"
+          echo "CHANGED_MODULES=$CHANGED_MODULES simulation" >> $GITHUB_ENV
+
+      - name: Update CHANGED_MODULES for world_modeling
+        if: steps.changed-files-world-modeling.outputs.any_changed == 'true'
+        run: |
+          echo "Detected world_modeling changes"
+          echo "CHANGED_MODULES=$CHANGED_MODULES world_modeling" >> $GITHUB_ENV
+
+      - name: Output list of changed modules
+        id: output-changes
+        run: |
+          echo "The changed modules are: $CHANGED_MODULES"
+          echo "modified_modules=$CHANGED_MODULES" >> $GITHUB_OUTPUT
+      
   setup-environment:
     name: Setup environment
     runs-on: ubuntu-latest
+    needs: get_changed_modules
+
+    env:
+      modified_modules: ${{ needs.get_changed_modules.outputs.modified_modules }}
```

Comment:
run inside setup-environment after turning the job into an action

Reply:
Sounds good, made the changes